app/models/metasploit_data_models/module_run.rb

```diff
@@ -0,0 +1,175 @@
+# making mischief on specific remote targets w/out the context of a vuln or even a remote IP service.
+#
+# There are also associations to {Mdm::Session} for two use cases: a `spawned_session` is a
+# session created by the ModuleRun. A `target_session` is a session that the ModuleRun
+# is acting upon (e.g.) for running a post module.
+class MetasploitDataModels::ModuleRun < ActiveRecord::Base
+  #
+  # Constants
+  #
+
+  # Marks the module as having successfully run
+  SUCCEED     = 'succeeded'
+  # Marks the run as having not run successfully
+  FAIL        = 'failed'
+  # Marks the module as having had a runtime error
+  ERROR       = 'error'
+  # {ModuleRun} objects will be validated against these statuses
+  VALID_STATUSES = [SUCCEED, FAIL, ERROR]
+
+
+  #
+  # Attributes
+  #
+
+  # @!attribute [rw] attempted_at
+  #   The date/time when this module was run
+  #   @return [Datetime]
+
+  # @!attribute [rw] fail_detail
+  #   Arbitrary information captured by the module to give in-depth reason for failure
+  #   @return [String]
+
+  # @!attribute [rw] fail_reason
+  #   One of the values of the constants in {Msf::Module::Failure}
+  #   @return [String]
+
+  # @!attribute [rw] module_name
+  #   The Msf::Module#fullname of the module being run
+  #   @return [String]
+
+  # @!attribute [rw] port
+  #   The port that the remote host was attacked on, if any
+  #   @return [Fixnum]
+
+  # @!attribute [rw] proto
+  #   The name of the protocol that the host was attacked on, if any
+  #   @return [String]
+
+  # @!attribute [rw] session_id
+  #   The {Mdm::Session} that this was run with, in the case of a post module. In exploit modules, this field will
+  #   remain null.
+  #   @return [Datetime]
+
+  # @!attribute [rw] status
+  #   The result of running the module
+  #   @return [String]
+
+  # @!attribute [rw] username
+  #   The name of the user running this module
+  #   @return [String]
+
+
+
+  #
+  # Associations
+  #
+
+
+
+  # @!attribute [rw] loots
+  #  The sweet, sweet loot taken by this module_run
+  #
+  #  @return [ActiveRecord::Relation<Mdm::Loot>]
+  has_many :loots,
+           class_name: 'Mdm::Loot',
+           inverse_of: :module_run
+
+
+  # @!attribute [rw] spawned_session
+  #
+  #  The session created by running this module.
+  #  Note that this is NOT the session that modules are run on.
+  #
+  #  @return [Mdm::Session]
+  has_one :spawned_session,
+             class_name: 'Mdm::Session',
+             inverse_of: :originating_module_run
+
+
+  # @!attribute [rw] target_session
+  #
+  #  The session this module was run on, if any.
+  #  Note that this is NOT a session created by this module run
+  #  of exploit modules.
+  #
+  #  @return [Mdm::Session]
+  belongs_to :target_session,
+             class_name: 'Mdm::Session',
+             foreign_key: :session_id,
+             inverse_of: :target_module_runs
+
+
+
+  # Declares this model to implement a polymorphic relationship with other models.
```

Comment:
It's not really an attribute. Afaik there's no way to use `trackable` outside of the polymorphism DSL in AR.